### PR TITLE
Log yamux session failure as notice

### PR DESCRIPTION
**Pull request by vihu**

### CODE REVIEW

File: src/libp2p_yamux_session.erl [hunk 1]
```diff
@@ -155,7 +155,9 @@ handle_info({inert_read, _, _}, State=#state{connection=Connection}) ->
             lager:notice("session closed"),
             {stop, normal, State};
         {error, Reason} ->
-            lager:error("Session header read failed: ~p ", [Reason]),
+            %% Apparently this happens a LOT
+            %% just flag a notice, don't pollute the logs with an error
+            lager:notice("Session header read failed: ~p ", [Reason]),
```
[the remark this answers is not in the document]
Comment (vihu) [reply src/libp2p_yamux_session.erl]: I was contemplating going down that path. Will talk to @madninja and filter unhelpful logs.

**Comment by vihu** [comment]
Just removed any logging for `enotconn` since that's logged like 90% of the times anyway.

File: src/libp2p_yamux_session.erl [hunk 2]
```diff
@@ -155,7 +155,15 @@ handle_info({inert_read, _, _}, State=#state{connection=Connection}) ->
             lager:notice("session closed"),
             {stop, normal, State};
         {error, Reason} ->
-            lager:error("Session header read failed: ~p ", [Reason]),
+            %% Apparently this happens a LOT
+            %% just flag a notice, don't pollute the logs with an error
+            case Reason == enotconn of
```
Comment (madninja) [inline src/libp2p_yamux_session.erl]: just do `{error, enotconn} -> ` and a next clause with `{error, reason}`. No need to nest case and have a boolean case statement  like this

Comment (vihu) [reply src/libp2p_yamux_session.erl]: Addresses: a07238d

**Pull request merged**